src/server-setup.ts

```diff
@@ -36,7 +37,8 @@ fastify.post<{ Body: UserRequest }>("/users", {
         birthDate: { type: 'string', format: 'date' }
       }
     }
-  }
+  },
+  preHandler: [authenticate]
```

Comment:
Uma dica para deixar o código mais claro seria criar um objeto para encapsular essa função lá no arquivo `authenticate-middleware`:

```suggestion
  preHandler: [AuthenticationMiddleware.authenticate]
```

Acho que assim fica mais claro para quem importa o midleware, o que acha?

Reply:
faz sentido